src/modules/gamewisp/subscriber-emotes.js

```diff
@@ -0,0 +1,205 @@
+const socketClient = require('./socket-client');
+const watcher = require('../../watcher');
+const twitch = require('../../utils/twitch');
+const settings = require('../../settings');
+
+const AbstractEmotes = require('../emotes/abstract-emotes');
+const Emote = require('../emotes/emote');
+
+const provider = {
+    id: 'gw-subscriber',
+    displayName: 'GameWisp Subscriber Emotes',
+    badge: 'https://d32y8axfzs6sv8.cloudfront.net/static/gamewisp_transparent_18px.png'
+};
+
+const buildEmote = ({id, channel, code, url}) => {
+    return new Emote({
+        id: id,
+        provider: provider,
+        channel: {name: channel},
+        code: code,
+        images: {
+            '1x': url,
+            '2x': url.replace('28x28', '56x56'),
+            '4x': url.replace('28x28', '112x112')
+        },
+        imgType: 'png'
+    });
+};
+
+class SubscriberEmotes extends AbstractEmotes {
+    constructor() {
+        super();
+
+        this.emoteUsers = new Map();
+
+        socketClient.on('initialize_room', data => this.initializeRoom(data));
+        socketClient.on('update_room', data => this.updateRoom(data));
+        socketClient.on('leave_room', data => this.leaveRoom(data));
+        socketClient.on('remove_emote', data => this.removeEmote(data));
+        socketClient.on('add_emote', data => this.addEmote(data));
+        socketClient.on('new_subscriber', data => this.newSubscriber(data));
+        socketClient.on('cancel_subscriber', data => this.cancelSubscriber(data));
+
+        watcher.on('load.chat', () => this.joinRoom());
+    }
+
+    get provider() {
+        return provider;
+    }
+
+    getEmotes(user) {
+        if (settings.get('gwEmotes') === false) return [];
+        if (!user) return [];
+
+        const emoteIDs = this.emoteUsers.get(user.name);
+        if (!emoteIDs) return [];
+
+        const emotes = [];
+
+        emoteIDs.forEach(id => {
```

Comment:
`id => this.emotes.has(id) && emotes.push(this.emotes.get(id))`

seems a bit cleaner